opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader.py

```diff
@@ -12,6 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+from abc import ABC, abstractmethod
+from typing import Callable, Iterable
 
-class MetricReader:
-    pass
+from typing_extensions import final
+
+from opentelemetry.sdk._metrics.point import AggregationTemporality, Metric
+
+_logger = logging.getLogger(__name__)
+
+
+class MetricReader(ABC):
+    def __init__(
+        self,
+        preferred_temporality: AggregationTemporality = AggregationTemporality.CUMULATIVE,
+    ) -> None:
+        self._collect: Callable[
+            ["MetricReader", AggregationTemporality], Iterable[Metric]
+        ] = None
+        self._preferred_temporality = preferred_temporality
+
+    @final
+    def collect(self) -> None:
+        """Collects the metrics from the internal SDK state and
+        invokes the `_receive_metrics` with the collection.
+        """
+        if self._collect is None:
+            _logger.warning(
+                "Cannot call collect on a MetricReader until it is registered on a MeterProvider"
+            )
+            return
+        self._receive_metrics(self._collect(self, self._preferred_temporality))
+
+    @final
+    def _set_collect_callback(
+        self,
+        func: Callable[
+            ["MetricReader", AggregationTemporality], Iterable[Metric]
+        ],
+    ) -> None:
+        self._collect = func
+
+    @abstractmethod
+    def _receive_metrics(self, metrics: Iterable[Metric]):
+        pass
+
+    @abstractmethod
+    def shutdown(self) -> bool:
+        """Shuts down the MetricReader. This method provides a way
+        for the MetricReader to do any cleanup required. A metric reader can
+        only be shutdown once, any subsequent calls are ignored and return
+        failure status.
+
+        Called when the SDK is shut down.
```

Comment:
nit maybe this is more clear

```suggestion
        When a `MetricReader` is registered on a `MeterProvider`, `MeterProvider.shutdown` will invoke this automatically.
```